### ISSUE TITLE
ClassCastException: java.lang.Integer cannot be cast to java.lang.String

### ISSUE THREAD

**Issue by kevin70** [spring-projects/spring-boot#11509]
java-version: 1.8
spring-boot-version: 1.5.7.RELEASE

application.yml
```
hello:
  123: world
```
There is no response to the above configuration startup program and no error information is seen in the console.

Exception stack:
```
java.lang.IllegalStateException: Failed to load property source from location 'classpath:/application.yml'
	at org.springframework.boot.context.config.ConfigFileApplicationListener$Loader.loadIntoGroup(ConfigFileApplicationListener.java:476)
	at org.springframework.boot.context.config.ConfigFileApplicationListener$Loader.load(ConfigFileApplicationListener.java:465)
	at org.springframework.boot.context.config.ConfigFileApplicationListener$Loader.load(ConfigFileApplicationListener.java:386)
	at org.springframework.boot.context.config.ConfigFileApplicationListener.addPropertySources(ConfigFileApplicationListener.java:225)
	at org.springframework.boot.context.config.ConfigFileApplicationListener.postProcessEnvironment(ConfigFileApplicationListener.java:195)
	at org.springframework.boot.context.config.ConfigFileApplicationListener.onApplicationEnvironmentPreparedEvent(ConfigFileApplicationListener.java:182)
	at org.springframework.boot.context.config.ConfigFileApplicationListener.onApplicationEvent(ConfigFileApplicationListener.java:168)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:127)
	at org.springframework.boot.context.event.EventPublishingRunListener.environmentPrepared(EventPublishingRunListener.java:74)
	at org.springframework.boot.SpringApplicationRunListeners.environmentPrepared(SpringApplicationRunListeners.java:54)
	at org.springframework.boot.SpringApplication.prepareEnvironment(SpringApplication.java:325)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:296)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:134)
	at com.memeyule.spark.server.ApplicationKt.main(Application.kt:198)
Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
	at org.springframework.beans.factory.config.YamlProcessor.buildFlattenedMap(YamlProcessor.java:276)
	at org.springframework.beans.factory.config.YamlProcessor.buildFlattenedMap(YamlProcessor.java:293)
	at org.springframework.beans.factory.config.YamlProcessor.buildFlattenedMap(YamlProcessor.java:293)
	at org.springframework.beans.factory.config.YamlProcessor.buildFlattenedMap(YamlProcessor.java:301)
	at org.springframework.beans.factory.config.YamlProcessor.buildFlattenedMap(YamlProcessor.java:293)
	at org.springframework.beans.factory.config.YamlProcessor.getFlattenedMap(YamlProcessor.java:270)
	at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:222)
	at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:162)
	at org.springframework.beans.factory.config.YamlProcessor.process(YamlProcessor.java:139)
	at org.springframework.boot.env.YamlPropertySourceLoader$Processor.process(YamlPropertySourceLoader.java:101)
	at org.springframework.boot.env.YamlPropertySourceLoader.load(YamlPropertySourceLoader.java:58)
	at org.springframework.boot.env.PropertySourcesLoader.load(PropertySourcesLoader.java:127)
	at org.springframework.boot.context.config.ConfigFileApplicationListener$Loader.doLoadIntoGroup(ConfigFileApplicationListener.java:490)
	at org.springframework.boot.context.config.ConfigFileApplicationListener$Loader.loadIntoGroup(ConfigFileApplicationListener.java:473)
```
After debugging, the exception was found, but it was not printed on the console, and the main thread was blocked.

**Thread Stack:**
```
"main" #1 prio=5 os_prio=0 tid=0x0000000002772800 nid=0xff78 waiting on condition [0x000000000271e000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000d7b766a8> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
	at org.springframework.boot.autoconfigure.BackgroundPreinitializer.onApplicationEvent(BackgroundPreinitializer.java:63)
	at org.springframework.boot.autoconfigure.BackgroundPreinitializer.onApplicationEvent(BackgroundPreinitializer.java:45)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:158)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:127)
	at org.springframework.boot.context.event.EventPublishingRunListener.finished(EventPublishingRunListener.java:115)
	at org.springframework.boot.SpringApplicationRunListeners.callFinishedListener(SpringApplicationRunListeners.java:79)
	at org.springframework.boot.SpringApplicationRunListeners.finished(SpringApplicationRunListeners.java:72)
	at org.springframework.boot.SpringApplication.handleRunFailure(SpringApplication.java:745)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:134)
	- locked <0x00000000d7a1fd88> (a java.util.concurrent.atomic.AtomicBoolean)
	at com.memeyule.spark.server.ApplicationKt.main(Application.kt:198)
```

-------------------------------------------------------------------------------------------------------------------

If the configuration is changed to run as follows
application.yml
```
hello:
  "123": world
```
In YML, key must be a string type.

**Comment by wilkinsona** [comment]
The hang and lack of diagnostics is a duplicate of #10362 and is fixed in 1.5.8. The integer key could, perhaps, be automatically converted to a string, but would have to be done in Spring Framework rather than Spring Boot. Please raised [a JIRA ticket](https://jira.spring.io/browse/SPR) if you’d like to pursue that possibility.
